doi/src/main/java/ca/nrc/cadc/doi/PostAction.java

```
@@ -463,11 +474,78 @@ private void performDoiAction() throws Exception {
             log.debug("redirectUrl: " + redirectUrl);
             syncOutput.setHeader("Location", redirectUrl);
             syncOutput.setCode(303);
+        } else if (doiAction.equals(SEARCH_ACTION)) {
+            if (doiStatusSearchFilter == null || callersNumericId == null) {
+                getStatusList(getOwnedDOIList());
+            } else {
+                getStatusList(getFilteredDOIList(doiStatusSearchFilter));
+            }
         } else {
             throw new UnsupportedOperationException("DOI action not implemented: " + doiAction);
         }
     }
-    
+
+    private List<Node> getFilteredDOIList(DoiStatusSearchFilter doiStatusSearchFilter) throws Exception {
+        List<Node> filteredNodes = new ArrayList<>();
+        ContainerNode doiRootNode = vospaceDoiClient.getContainerNode("");
+
+        if (doiRootNode != null) {
+            for (Node childNode : doiRootNode.getNodes()) {
+                NodeProperty requester = childNode.getProperty(DOI_VOS_REQUESTER_PROP);
+                if (requester == null || requester.getValue() == null) {
```

Every node should have a requester property. If it doesn't, we should log an error message saying so. And I wonder if we should throw an exception. Only logging the error hides the problem, and we don't generally check error logs for problems. But throwing an exception would effectively disable search if one node was missing a requestor property, so maybe logging the error is correct.

---

In the case of DOI, we will have a node for rafts. So if we throw 'exception' or log 'error', it will technically be wrong. I can log error if there can be nothing other than doi's container nodes and we are expecting all the child nodes with requester property. Please suggest.

---

A DOI parent node not having a requester nodeProperty is an error, it's just how much of an error. It shouldn't happen, but what's the proper way to handle it if it does? Throwing an exception would make the doi or rafts service unusable. It the requestor property is missing, log an error and move on to the next.